### ISSUE TITLE
Espresso server - spawn EINVAL Gradle error message

### ISSUE THREAD

**Issue by pbdeguzman** [appium/appium-espresso-driver#996]
Hi, I'm new here and I tried to use Appium Espresso Driver as my Appium Framework on my Appium Inspector.

- I run Appium via Command Line.

![image](https://github.com/appium/appium-espresso-driver/assets/42160425/cb8f016e-6438-41a7-9f14-86761c0b7657)

- Then I setup the Appium Inspector and Start Session. Then, I encounter this error.

![image](https://github.com/appium/appium-espresso-driver/assets/42160425/383dfaba-585c-46b2-9c77-5ae664e38fab)

- I check my test device and the espresso driver has been installed
![image](https://github.com/appium/appium-espresso-driver/assets/42160425/71cf4225-7e58-470d-82f3-4c7fa1fcd189)

Here's my logs:
[logs.txt](https://github.com/appium/appium-espresso-driver/files/15019855/logs.txt)

Is my setup correct? Or is there something missing on my end? Any help would be appreciated.

Thanks,

Patrick

**Comment by KazuCocoa** [comment]
I would try out the gradle command directly. Perhaps the espresso server dependencies needs to be coordinated. It is not only building step but also launching the test app via instrument process (in espresso framework) related

**Comment by mykola-mokhnach** [comment]
Seems related to https://github.com/appium/appium/issues/20008

**Comment by KazuCocoa** [comment]
The latest issue was your espresso server and the app under test did not have the same signature.
```
[EspressoDriver@6f23]   "alwaysMatch": {
[EspressoDriver@6f23]     "platformName": "Android",
[EspressoDriver@6f23]     "appium:udid": "0123456789ABCDEF",
[EspressoDriver@6f23]     "appium:deviceName": "p2pro",
[EspressoDriver@6f23]     "appium:automationName": "Espresso",
[EspressoDriver@6f23]     "appium:appPackage": "com.global.wwca",
[EspressoDriver@6f23]     "appium:forceEspressoRebuild": "true",
[EspressoDriver@6f23]     "appium:showGradleLog": "true",
[EspressoDriver@6f23]     "appium:espressoBuildConfig": "{\"additionalAndroidTestDependencies\": [\"androidx.lifecycle:lifecycle-extensions:2.2.0\", \"com.google.android.material:material:1.6.1\",  \"androidx.constraintlayout:constraintlayout:2.1.4\"]}",
[EspressoDriver@6f23]     "appium:ensureWebviewsHavePages": true,
[EspressoDriver@6f23]     "appium:nativeWebScreenshot": true,
[EspressoDriver@6f23]     "appium:newCommandTimeout": 3600,
[EspressoDriver@6f23]     "appium:connectHardwareKeyboard": true
[EspressoDriver@6f23]   },
```

> Both the server package and the application under test must be signed with the same digital signature.

https://github.com/appium/appium-espresso-driver?tab=readme-ov-file#requirements

for example https://github.com/appium/appium-espresso-driver?tab=readme-ov-file#app-signing would help, or use `app` capability instead of appPackage etc...

**Comment by KazuCocoa** [comment]
(closing as the original issue was fixed by the #997)
